app/src/main/java/com/aaronjwood/portauthority/activity/MainActivity.java

```diff
@@ -146,16 +146,16 @@ public void onClick(View v) {
 
                 hosts.clear();
                 hostsAdapter.notifyDataSetChanged();
-
+                int numberOfHosts = wifi.getNumberOfHostsInWifiSubnet();
                 scanProgressDialog = new ProgressDialog(MainActivity.this, R.style.DialogTheme);
                 scanProgressDialog.setCancelable(false);
                 scanProgressDialog.setTitle("Scanning For Hosts");
                 scanProgressDialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
                 scanProgressDialog.setProgress(0);
-                scanProgressDialog.setMax(255);
+                scanProgressDialog.setMax(numberOfHosts);
                 scanProgressDialog.show();
 
-                Discovery.scanHosts(wifi.getInternalWifiIpAddress(), MainActivity.this);
+                Discovery.scanHosts(wifi.getInternalWifiIpAddress(), Integer.toString(numberOfHosts), MainActivity.this);
```

Comment:
You should be able to get rid of the `Integer.toString()` since you `decode()` it later on in `ScanHostsAsyncTask`.